virtcontainers/sandbox.go

```diff
@@ -1923,9 +1926,30 @@ func (s *Sandbox) calculateSandboxMemory() int64 {
 
 		if m := c.Resources.Memory; m != nil && m.Limit != nil {
 			memorySandbox += *m.Limit
+			s.Logger().WithField("memory limit", memorySandbox).Info("Memory Sandbox + Memory Limit ")
+		}
+
+		//Add hugepages memory
+		//HugepageLimit is uint64 - https://github.com/opencontainers/runtime-spec/blob/master/specs-go/config.go#L242
+		for _, l := range c.Resources.HugepageLimits {
+			if l.Limit <= math.MaxInt64 {
+				memorySandbox += int64(l.Limit)
+			} else {
+				//Overflow
+				s.Logger().WithField("hugepages memory limit overflow", l.Limit).Warn("Memory Sandbox + Hugepages Memory Limit ")
+				return 0, errors.New("Integer overflow")
+			}
+
+			s.Logger().WithField("hugepages memory limit", memorySandbox).Info("Memory Sandbox + Hugepages Memory Limit ")
 		}
 	}
-	return memorySandbox
+
+	//Check overflow
+	if memorySandbox < 0 {
+		return 0, errors.New("Integer overflow")
+	} else {
+		return memorySandbox, nil
```

Comment:
From taking a look, I think it may be worth clarifying here that we don't differentiate between 'normal pages' and hugePages, since the entire VM will be backed by huge pages.  

Do I understand it correctly?  I think this is probably a higher-level way to validate that we're doing the right thing here, too (ie, do we error if the VM isn't backed by huge pages).

Reply:
> From taking a look, I think it may be worth clarifying here that we don't differentiate between 'normal pages' and hugePages, since the entire VM will be backed by huge pages.
> 
> Do I understand it correctly? I think this is probably a higher-level way to validate that we're doing the right thing here, too (ie, do we error if the VM isn't backed by huge pages).

Yeah true.. We don't care if the VM is backed by hugepages or not.  Now question is should we allow hugepages inside the Kata VM only if the entire VM is backed by hugepages or we keep the two separately ?  

Although having hugepages inside the VM without the VM memory itself being backed by hugepages will not be of much use other than ensuring the apps requiring hugepages continues to run on Kata (eg. dpdk apps).